### ISSUE TITLE
Add support for spherecast

### ISSUE THREAD

**Issue by yaustar** [playcanvas/engine#2094]
From forum thread: https://forum.playcanvas.com/t/ability-to-use-spherecast/13310/3

It would be great to add some form of sphere casting if the current Ammo,js API allows it.

**Comment by albertvanveen** [comment]
I would also like to see this feature in PlayCanvas because I could use it in different situations.

**Comment by Phuongstudent** [comment]
Yes I would love this feature as well as it would prevent me from having multiple raycasts

**Comment by simonbq** [comment]
This would be a great feature as it would make it easier to improve click accuracy on mobile devices instead of raycasting.

**Comment by LeXXik** [comment]
@simonbq the Ammo version that PC currently only has partial support for it. I made an extension for convex shapes sweep tests. You can grab a later version of Ammo from there too, which enables full support.

https://forum.playcanvas.com/t/playcanvas-physics-extension/13737

**Comment by simonbq** [comment]
@LeXXik Cheers, I'll check it out. I'm mainly interested in it being added to the hosted version of PlayCanvas though.

**Comment by keslyh** [comment]
This will be useful for a camera collision system, it will make it smoother and not punch through certain size gaps.

**Comment by LeXXik** [comment]
@yaustar @willeastcott I could add a support for convex sweeptests, but the current Ammo version needs to be updated.

**Comment by willeastcott** [comment]
@LeXXik What would the newly introduced PlayCanvas API be? I think it's fine if the ammo build needs to be updated. You'd just check for the necessary functions and if they're not there, just print a warning.

**Comment by LeXXik** [comment]
Well, we already have the following raycast methods on `rigidbody` component system:
```
app.systems.rigidbody.raycast(...);
app.systems.rigidbody.raycastAll(...);
```
I do think they belong under `collision` component system, though, as rigid bodies are shapeless and not able to collide. It is a collision component that gives body a shape, mass, and other properties that rays can detect. In fact, own `physics` system would be even better, which would handle casts in the physics world of the current backend, but that can be left for a better day. Nevertheless, we can add similar:
```
app.systems.rigidbody.shapeCast(...);
```
As with `raycast` it would take starting position and end position of the cast, as well as required shape that would be used for the sweep test. It can optionally take starting and final rotation quaternions of the shape.

Because the shape is an Ammo object, I did have convenience methods, like `.sphereCast()` or `.boxCast()`, etc, which would create the shape for the user and destroy it when finished. To simplify the API, we can add enums, like `pc.SHAPECAST_SPHERE` that `shapeCast()` method would take with optional params that describe it, like sphere shape radius, for example, etc.

I will make a draft PR.
